OptimizelySDK/Customization/DefaultEventDispatcher.swift

```diff
@@ -75,9 +75,9 @@ open class DefaultEventDispatcher: BackgroundingCallbacks, OPTEventDispatcher {
     open func dispatchEvent(event: EventForDispatch, completionHandler: DispatchCompletionHandler?) {
         dataStore.save(item: event)
         
-        // TODO: use or clean up completionHandler
-
         setTimer()
+        
+        completionHandler?(.success(event.body))
```

Comment:
this doesn't seem like the correct approach.  maybe something  better is to store a list of all callbacks without a retain and then call them on anytime an event is actually sent.

Reply:
The current way of calling back on queueing makes sense to me:
- We do not have any case for queueing failure for now, but this call can be used for notification of "event queued successfully and it'll be delivered ok"
- Event delivery can be delayed arbitrary time. Keeping all callback handlers may be expensive and error-prone.